### PR TITLE
NCC: Life of the Party and support

### LINKED ISSUE #379: Life of the Party

### PR DESCRIPTION
**Pull request by Northmoc**
support for Goad that lasts the rest of the game

closes #379

### CODE REVIEW

**Review by tool4ever — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+PT:0/1
+K:First Strike
+K:Trample
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigPump | TriggerDescription$ Whenever CARDNAME attacks, it gets +X/+0 until end of turn, where X is the number of creatures you control.
+SVar:TrigPump:DB$ Pump | NumAtt$ X
+SVar:X:Count$Valid Creature.YouCtrl
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self+nonToken | Execute$ TrigCopyPermanent | TriggerDescription$ When CARDNAME enters the battlefield, if it's not a token, each opponent creates a token that's a copy of it. The tokens are goaded for the rest of the game. (They attack each combat if able and attack a player other than you if able.)
+SVar:TrigCopyPermanent:DB$ RepeatEach | RepeatSubAbility$ DBClone | RepeatPlayers$ Opponent | ChangeZoneTable$ True | SubAbility$ DBGoad
+SVar:DBClone:DB$ CopyPermanent | Defined$ TriggeredCard | Controller$ Remembered | ImprintTokens$ True
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt:13]: imo for later effects what do you guys think about extending CopyPermanent when using multiple "Controller", that it just creates multiple Tokens for them? so reducing the amount of RepeatEach?

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt:13]: for ReplacementEffects, the token might be created at the same time

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt:13]: Ah good idea 👍 
I forgot I already did a part for that

`ForEach$ Opponent` should be almost working
just need to add another param there so it uses the current player as token owner

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt:13]: i don't think "CopyPermanent" has ForEach Param. i would have just used Controller instead

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt:13]: I meant I already added it

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt:13]: nono ForEach is for when the Creature needs to remember the Opponent

this card needs to be created for each opponent under opponents control, so thats why `Controller` needs to be extended

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt]: What do you two think of `EachCreates` ?

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt]: why not just extending the `Controller` Param to work with a List of Players instead?

we don't need an extra Parameter for this

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/life_of_the_party.txt]: You're right, of course

I just became nervous due to all the times Controller is accessed throughout the resolve

**Review by Hanmac — APPROVED** [review]
:+1:

**Pull request merged**